ogc/src/main/scala/geotrellis/server/ogc/OgcTime.scala

```diff
@@ -17,9 +17,53 @@
 package geotrellis.server.ogc
 
 import jp.ne.opt.chronoscala.Imports._
+import cats.data.NonEmptyList
+import cats.Semigroup
+import cats.syntax.semigroup._
 
 import java.time.ZonedDateTime
 
+sealed trait OgcTime {
+  def isEmpty: Boolean = false
+  def nonEmpty: Boolean = !isEmpty
+}
+
+case object OgcTimeEmpty extends OgcTime {
+  override def isEmpty: Boolean = true
+  implicit val ogcTimeEmptySemigroup: Semigroup[OgcTimeEmpty.type] = Semigroup.instance { (l, _) => l }
+}
+
+object OgcTime {
+  implicit val ogcTimeSemigroup: Semigroup[OgcTime] = Semigroup.instance {
+    case (l: OgcTimePositions, r: OgcTimePositions) => l |+| r
+    case (l: OgcTimeInterval, r: OgcTimeInterval) => l |+| r
+    case (l: OgcTimePositions, _: OgcTimeEmpty.type) => l
+    case (l: OgcTimeInterval, _: OgcTimeEmpty.type) => l
+    case (_: OgcTimeEmpty.type, r: OgcTimePositions) => r
+    case (_: OgcTimeEmpty.type, r: OgcTimeInterval) => r
+    case (l, _) => l
+  }
+}
+
+final case class OgcTimePositions(list: NonEmptyList[ZonedDateTime]) extends OgcTime {
```

Comment:
The comma separated format described in #259 could be described by `List[OgcTimeInterval]` where only the `start` property is set for each `OgcTimeInterval`. In effect, `List[OgcTimePositions]` is the same as `List[OgcTimeInterval(start, None, None)]`, and we should be able to parse a string containing comma separated single times, comma separated intervals and a single interval all into the same `List[OgcTimeInterval]` type and then render them back out to XML based on the specifics of the WCS or WMS spec on the other side.

As a result, I'm curious as to why we now have two diverging types `OgcTimePositions` and `OgcTimeInterval`? What am I missing where the existing types weren't sufficient? I'm trying to avoid us adding additional code complexity where possible for an already complex topic.

Reply:
@CloudNiner I decided to add  richer semantics to simplify the code.

Reply:
`OgcTime` is an ADT now, wich can be an interval or can be a list of times that probably are not an interval. For sure Positions can be represented as `OgcTimeInterval(start, None, None)` but I don't think it is much more readable and the type can add semantics.

Reply:
Gotcha. Then in the absence of this change giving us some feature that we didn't have before, I have to disagree with this change. 

IMO it has cost us maintainability in that as a programmer looking to work on this library I have to unpack when and where I should use each OgcTime type when. And if I need a new type I have to add it everywhere and have a strong understanding what the code is doing at each point to implement correctly. We also are missing the case where a list of OgcTime could contain position and interval intermixed -- the implementation of that would be an additional OgcTime as `List[OgcTimeInterval]` which is just a general case of all the currently implemented OgcTime types.

A better way to apply some sort of ADT here would be to do it at the individual service level. For example, if WCS wants to explicitly support a lists of positions, then have something like a WCSTimePositions specific to that service rather than complicating the current `List[OgcTimeInterval]` data structure that is defined as described in the ogc spec.

Or, if the core change here is that we need an `OgcSource` to support a list of times, then we should just change `def timeInterval: Option[OgcTimeInterval]` to `def times: List[OgcTimeInterval]`. Now that I think about it, I'm not sure why I didn't do that with the original implementation.

Reply:
@CloudNiner WMS also supports TimePoints